### PR TITLE
Fix CopyToClipboard issue

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: #599 

In this PR, I added the copyToClipboard action and update CopyToClipboardAction

### CODE REVIEW

File: lib/framework/action.dart [hunk 1]
```diff
@@ -445,6 +445,18 @@ class CopyToClipboardAction extends EnsembleAction {
   String? value;
```
Comment (vusters) [inline lib/framework/action.dart:445]: what if it's an integer? use dynamic here

File: lib/framework/action.dart [hunk 2]
```diff
@@ -445,6 +445,18 @@ class CopyToClipboardAction extends EnsembleAction {
...
   EnsembleAction? onSuccess;
   EnsembleAction? onFailure;
+
+  factory CopyToClipboardAction.fromYaml({YamlMap? payload}) {
+    if (payload == null || payload['value'] == null) {
+      throw LanguageError(
+          '${ActionType.copyToClipboard.name} requires the value.');
+    }
+    return CopyToClipboardAction(
+      value: Utils.optionalString(payload['value']),
```
Comment (vusters) [inline lib/framework/action.dart:455]: `value: payload['value']`

File: lib/screen_controller.dart [hunk 3]
```diff
@@ -386,11 +386,14 @@ class ScreenController {
       }
     } else if (action is CopyToClipboardAction) {
       if (action.value != null) {
-        Clipboard.setData(ClipboardData(text: action.value!)).then((value) {
-          if (action.onSuccess != null) {
-            executeAction(context, action.onSuccess!);
-          }
-        });
+        String? clipboardValue = dataContext.eval(action.value);
```
Comment (vusters) [inline lib/screen_controller.dart:389]: can we abstract out the eval and do it inside the Action. See StartTimerAction's `getStartAfter()`

**Review by vusters — APPROVED** [review]
This is a problem with all Actions and I have been trying to refactor them slowly (see StartTimerAction). It's time to fix this for all Actions otherwise we'll keep running into this. 
Created a separate ticket for you Vinoth https://github.com/orgs/EnsembleUI/projects/2/views/1?pane=issue&itemId=31776764

File: lib/screen_controller.dart (hunk 3 above)
Comment (vinothvino42) [reply lib/screen_controller.dart]: Refactored it

File: lib/framework/action.dart (hunk 2 above)
Comment (vinothvino42) [reply lib/framework/action.dart]: Done

File: lib/framework/action.dart (hunk 1 above)
Comment (vinothvino42) [reply lib/framework/action.dart]: Changed it to dynamic

**Review by vusters — APPROVED** [review]

**Pull request merged**